include/revng/Storage/StorageClient.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <memory>
+
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/Error.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/Path.h"
+
+#include "revng/Support/Assert.h"
+#include "revng/Support/Debug.h"
+
+namespace revng {
+
+class WritableFile {
```

Comment:
Given that you created `librevngStorage`, move `WritableFile` and `ReadableFile` to their own file.